cmd/soapboxd/main.go

```diff
@@ -118,3 +123,56 @@ func timingInterceptor(
 	log.Printf("method=%s duration=%s error=%v", info.FullMethod, time.Since(t0), err)
 	return resp, err
 }
+
+func loginInterceptor(
+	ctx context.Context,
+	req interface{},
+	info *grpc.UnaryServerInfo,
+	handler grpc.UnaryHandler,
+) (interface{}, error) {
+	if md, ok := metadata.FromContext(ctx); ok {
+		if _, ok := md["skip_auth"]; ok {
```

Comment:
I think fundamentally, we can't let the client be the one deciding whether an RPC requires auth or not -- it would be too trivial for a malicious client to bypass auth on sensitive calls.

I think instead, each RPC implementation on the server-side should indicate somehow that it requires auth. Then the interceptor here would test to see if a given handler required auth, and then would call `authorize`.

Off the top of my head, I'm not sure what the best way is to do this. One challenge is that the methods have different signatures, so a naive approach of creating a type that defined them, and then mapping from that type to a bool (whether auth is required), would not be straightforward. You could do a map of `interface{}` to `bool`, type-switching on each method, but this could be very verbose. I think we need to think about how to do this before proceeding. There are probably a couple of different approaches. I am happy to help think through this.

Reply:
Ahh, that is a great point. I think I was just trying to get something working, but we definitely need to have the server deciding when to auth rather than the client. Let me see if I can get something working here, if I get stuck I'll reach out.

Reply:
There are only two server methods that don't need to be authed, namely `LoginUser` and `CreateUser`, so if we can read the method names in the interceptor somehow, we can just skip for those methods. Not sure how to read which method is being called from there though, is it listed in the Context?

Reply:
The `info` param to that function, of type `UnaryServerInfo`, is a struct type that has a field `FullMethod` which is the full RPC method string -- you could do some pattern-matching on that.